src/ORM/DataObjectSchema.php

```diff
@@ -902,12 +920,36 @@ public function hasOneComponent($class, $component)
             return null;
         }
 
+        $spec = $hasOnes[$component];
+
         // Validate
-        $relationClass = $hasOnes[$component];
+        if (is_array($spec)) {
+            $this->checkHasOneArraySpec($class, $component, $spec);
+        }
+        $relationClass = is_array($spec) ? $spec['class'] : $spec;
         $this->checkRelationClass($class, $component, $relationClass, 'has_one');
+
         return $relationClass;
     }
 
+    /**
+     * Check if a has_one relation handles multiple reciprocal has_many relations.
+     *
+     * @return bool True if the relation exists and handles multiple reciprocal has_many relations.
+     */
+    public function hasOneComponentHandlesMultipleHasMany(string $class, string $component): bool
```

Comment:
```suggestion
    public function hasOneComponentHandlesMultipleRelations(string $class, string $component): bool
```

Isn't this method misnamed? I thought that the multi-reciprocal has_one could also be used on as a 'doubled-up has_one' so that we can go Link->Owner() and it find owning Page, whether or not the relation to Link on the Page is a has_one or a has_many?

Reply:
>  I thought that the multi-reciprocal has_one could also be used on as a 'doubled-up has_one' so that we can go Link->Owner() and it find owning Page, whether or not the relation to Link on the Page is a has_one or a has_many?

That's a byproduct, and is something we're implementing a little bit of logic into linkfield explicitly to allow. "handlesMultipleRelations" implies that out of the box you can point any relation at this has_one and it'll be handled. That isn't the case - that only works with multiple has_many relations.

That said, it _could_ theoretically be extended to work with multiple belongs_to relations as well... so I'll update the name on that basis.

Reply:
Done